### ISSUE TITLE
Controller command errors

### ISSUE THREAD

**Issue by pastukhov** [OpenZWave/node-red-contrib-openzwave#13]
Time to time i see an events with topic 'zwave: controller command'  and payload
```JSON
{
    "state" : 0,
    "errcode" : 1,
}
```
Error codes, i see, can be 1,6 or 7.
Whait is this?
I think this is should reflected in readme.

**Comment by ekarak** [comment]
It is documented. Look for BeginControllerCommand in management README

Στις 12 Νοεμβρίου 2015 9:20:01 π.μ. EET, ο/η Artem Pastukhov <notifications@github.com> έγραψε:
>Time to time i see an events with topic 'zwave: controller command' 
>and payload
>```JSON
>{
>    "state" : 0,
>    "errcode" : 1,
>}
>```
>Error codes, i see, can be 1,6 or 7.
>Whait is this?
>I think this is should reflected in readme.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/OpenZWave/node-red-contrib-openzwave/issues/13

-- 
Sent from my mobile device with K-9 Mail. Please excuse my brevity.

**Comment by pastukhov** [comment]
Thanks!

**Comment by pastukhov** [comment]
Are you talking about https://github.com/OpenZWave/node-openzwave-shared/blob/master/README-mgmt.md ?
There is no error codes description.
I foud https://github.com/OpenZWave/node-openzwave-shared/blob/master/README-events.md, but there is no error codes description. For example, what errcode  7 means?

**Comment by ekarak** [comment]
I've just committed a set of changes in node-openzwave-shared. Please try compiling ('node-gyp rebuild') the 'buttons' branch (not master!), and fetch latest code from this repository. It adds helpful text messages in NR notifications.

**Comment by ekarak** [comment]
resolving - this is not an issue with the NR adapter, but with node-openzwave-shared. Please reopen a new issue, mentioning this one, in the node-openzwave-shared if you still have problems.

**Comment by pastukhov** [comment]
I'm getting ```[zwave] Error: /opt/node-red/.node-red/node_modules/node-red-contrib-openzwave/node_modules/openzwave-shared/build/Release/openzwave_shared.node: undefined symbol: _ZN3OZW20ozw_watcher_callbackEPKN9OpenZWave12NotificationEPv```
with 'buttons' branch

**Comment by ekarak** [comment]
You're right. This is a bug in OpenZWave. Going to submitting a patch.

**Comment by ekarak** [comment]
https://github.com/OpenZWave/open-zwave/pull/693

**Comment by pastukhov** [comment]
Now i have
```
 [zwave] Error: /opt/node-red/.node-red/node_modules/node-red-contrib-openzwave/node_modules/openzwave-shared/build/Release/openzwave_shared.node: undefined symbol: _ZN3OZW14ozw_vers_majorE
```

**Comment by ekarak** [comment]
Artem can you please 'git pull' the buttons branch and try again?

**Comment by pastukhov** [comment]
```
nm: /usr/lib/arm-linux-gnueabihf//libopenzwave.so: no symbols
gyp: Call to 'nm /usr/lib/arm-linux-gnueabihf//libopenzwave.so | grep _ZN9OpenZWave7Manager7AddNodeEjb | wc -l' returned exit status 0. while trying to load binding.gyp
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/usr/lib/node_modules/node-gyp/lib/configure.js:355:16)
gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1074:12)
gyp ERR! System Linux 4.2.0-bananian
gyp ERR! command "node" "/usr/bin/node-gyp" "rebuild"
gyp ERR! cwd /opt/node-red/.node-red/node_modules/node-red-contrib-openzwave/node_modules/openzwave-shared
gyp ERR! node -v v0.12.7
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok 
```

**Comment by ekarak** [comment]
OK sorry about this.  I've rolled back this experimental symbol mangling stuff and released version 1.1.5 of node-openzwave-shared. You'll see that there is an extra argument passed to the event callbacks (helpmsg) which should contain a helpful string of what is the meaning of the controller response.

**Comment by ekarak** [comment]
Released version 1.1.5 of the node-red plugin, which should hopefully include a 'help' string in the 'controller command' message in the NR flow:
```
        function controllerCommand(nodeid, state, errcode, help) {
                if (debug) console.log('ZWave controller command feedback received');
                zwcallback('controller command', {nodeid: nodeid, state: state, errcode: errcode, help: help});
        }
```

**Comment by pastukhov** [comment]
Thanks!
